lib/kubeclient/common.rb

```diff
@@ -127,8 +127,11 @@ def self.define_entity_methods(entity_types)
     end
 
     def self.pluralize_entity(entity_name)
-      return entity_name + 's' if entity_name.end_with? 'quota'
-      entity_name.pluralize
+      entity_name.end_with?('s') ? entity_name + 'es' : entity_name + 's'
+    end
+
+    def self.underscore_entity(entity_name)
+      entity_name.gsub(/([a-z])([A-Z])/, '\1_\2').downcase
```

Comment:
This method has to handle very specific, known in advance, values. I've added a test with supported values.